### PR TITLE
ddl: Add telementry for distributed reorg tasks.

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/41266

Problem Summary:
Add telementry counter to get distrbuted reorg DDL tasks statistics.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

```release-note
None
```

### CODE REVIEW

File: telemetry/data_feature_usage_test.go [hunk 1]
```diff
@@ -576,6 +576,36 @@ func TestAddIndexAccelerationAndMDL(t *testing.T) {
...
 
+func TestDistReorgUsage(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	usage, err := telemetry.GetFeatureUsage(tk.Session())
+	require.Equal(t, int64(0), usage.DDLUsageCounter.DistReorgUsed)
+	require.NoError(t, err)
+
+	allow := variable.DDLEnableDistributeReorg.Load()
+	require.Equal(t, false, allow)
```
Comment (zimulala) [inline telemetry/data_feature_usage_test.go:587]: We are not sure what the value is, depending on what you set it up to when you run the test, so you have to decide

Comment (Benjamin2037) [reply telemetry/data_feature_usage_test.go:587]: In 6.6， this system variable is default false right?

Comment (zimulala) [reply telemetry/data_feature_usage_test.go:587]: Yes, but in CI we will run it with `distributereorg` is true and false.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

File: metrics/telemetry.go [hunk 2]
```diff
@@ -169,6 +169,13 @@ var (
 			Name:      "compact_partition_usage",
 			Help:      "Counter of compact table partition",
 		})
+	TelemetryDistReorgCnt = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "telemetry",
+			Name:      "Distributed_reorg_count",
```
Comment (hawkingrei) [inline metrics/telemetry.go:176]:
```suggestion
			Name:      "distributed_reorg_count",
```

metrics's name should be lower case.

**Pull request merged**